### DISCUSSION TITLE
How should flat config plugins handle recommended configs with different sub-sets for multiple file types?

### DISCUSSION THREAD

**Discussion by Standard8** [eslint/eslint#18576] [Q&A] [answered]
I have a plugin that I am looking to upgrading to flat config where different rules are applied based on the file type. For example, the existing plugin looks something like:

```js
overrides: [{
  files: ["*.mjs"], rules: { ...some rules... },
  files: ["*.jsx"], rules: { ...some other rules... },
  files: ["*.worker.js"], rules { ...more different rules... },
}],
rules: {
  // main rules
},
```

What's the correct way to handle this in the flat configuration world? It looks like I can specify something like:

```js
const plugin = {
    meta: {},
    configs: {},
    rules: {},
    processors: {}
};

Object.assign(plugin.configs, {
  "flat/recommended": {
    plugins: { myplugin: plugin },
    files: "*.mjs",
    rules: {
      // some rules
    }
  }
});
```
However, that would obviously only handle one file type per configuration.

Would the work around for this be to export multiple configs - one for each pattern, and then maybe an additional wrapper that's actually an array?

**Comment by nzakas** [comment, marked as answer]
> Would the work around for this be to export multiple configs - one for each pattern, and then maybe an additional wrapper that's actually an array?

Yes. If your intent is to have all of the configs used together 100% of the time, then just export an array. If your intent is to allow people to pick and choose, then exporting individual configs plus an array config makes sense.
